### PR TITLE
minor changes in the project & basic functionality for yeelight ceiling

### PR DESCRIPTION
**Pull request by dimalo**

### CODE REVIEW

**Comment by dimalo** [comment]
for me it works out addings the channels in the abstract handler in defineDeviceType().
removed channeltypeUID, I believe its useless - they get set without doing anything.
now I also get the behaviour with the actions channel group disappearing.
Seems like the thingbuilder is missing a .withChannelGroup...

I can imagine you have something else in mind so its your call to pull the last commit. the other two should be ok...

File: addons/binding/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/handler/MiIoAbstractHandler.java [hunk 1]
```diff
@@ -342,4 +357,68 @@ protected JsonObject getJsonResultHelper(String response) {
         }
         return null;
     }
+
+    private boolean buildChannelStructure(String deviceName) {
```
Comment (marcelrv) [inline addons/binding/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/handler/MiIoAbstractHandler.java]: Why moving this to to abstract handler.
E.g the vacuumhandler will not need this at all.

My intention is to have the basic handler for the devices that follow the basic approach properties via getprop command and simple actions.
More special devices like the vacuum have their own handler extending the abstract handler

**Comment by dimalo** [comment]
I thought so 😄  That was the quickest fix to overcome the static modelID in buildChannelStructure()...

File: addons/binding/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/handler/MiIoAbstractHandler.java (hunk 1 above)
Comment (dimalo) [inline addons/binding/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/handler/MiIoAbstractHandler.java]: I didn't think you are planning on implementing special handlers at all.
Even then it's not bad to have it in place with the thing definition.
But like I said, it's your call 👍

**Comment by marcelrv** [comment]
I see, there are few ways, one option I was considering was to make it part of the configuration.
Than it is available from the start, or when not available yet, you just don't make the channel config.
(somewhat similar as how the deviceID is done)
Hence the init procecess
stage 1) getting tolken
stage 2) getting the deviceID
stage3) getting the deviceType. Ones the 3rd is available, channels can be configured.

Using that, also helps with new devices (& testing). For an unsupported / undefined one, you can than 'mimic' an already existing device by updating the modelparameter.

**Pull request merged**